src/util/misc/projectStroke.ts

```diff
@@ -39,62 +49,216 @@ export const projectStrokeOnPoints = (
+      } else {
+        adjustedStrokeMiterLimit = strokeMiterLimit
+      }
+
+      if (hypot(miterVector.x, miterVector.y) / s <= adjustedStrokeMiterLimit) {
+        const proj1 = applySkew(A.add(miterVector));
+        coords.push({
+          "projectedPoint": proj1,
+          "originPoint": A,
+          "bisector": bisector
+        });
         return;
       }
     }
-    scalar = -s * Math.SQRT2;
-    miterVector = new Point(
-      bisectorVector.x * scalar * strokeUniformScalar.x,
-      bisectorVector.y * scalar * strokeUniformScalar.y
-    );
-    coords.push(A.add(miterVector));
-    coords.push(A.subtract(miterVector));
+    /* ROUND (without skew)
+      Calculation: the projections are the two vectors parallel to X and Y axes
+      Visually detailed here: https://github.com/fabricjs/fabric.js/pull/8083
+    */
+    if (strokeLineJoin === 'round' && skewX === 0 && skewY === 0) {
+      // correctSide is used to only consider projecting for the outer side 
+      const correctSide = new Point(
+          Math.abs(Math.atan2(bisectorVector.y, bisectorVector.x)) >= PiBy2 ? 1 : -1, 
+          Math.abs(Math.atan2(bisectorVector.x, bisectorVector.y)) >= PiBy2 ? 1 : -1
+        ),
+        radiusOnAxisX = new Point(s * strokeUniformScalar.x, 0).multiply(correctSide), 
+        radiusOnAxisY = new Point(0, s * strokeUniformScalar.y).multiply(correctSide),
+        proj1 = applySkew(A.add(radiusOnAxisX)), 
+        proj2 = applySkew(A.add(radiusOnAxisY));
+      [proj1, proj2].forEach(proj => {
+        coords.push({
+          "projectedPoint": proj,
+          "originPoint": A,
+          "bisector": bisector
+        });
+      });
+    } 
+    /* ROUND (with skew)
+      Calculation: the projections are the points furthest from the vertex in the direction of the X and Y axes after distortion
+      Visually detailed here: https://github.com/fabricjs/fabric.js/pull/8083
+    */
+    else if (strokeLineJoin === 'round'&& (skewX !== 0 || skewY !== 0)) {
+      // We calculate the start and end points of the circle segment
+      const AB = createVector(strokeUniform ? A.multiply(scale) : A, strokeUniform ? B.multiply(scale) : B), 
+          AC = createVector(strokeUniform ? A.multiply(scale) : A, strokeUniform ? C.multiply(scale) : C);
+
+      [AB, AC].forEach(function (vector) {
+        const hatOrthogonal = getOrthogonalUnitVector(vector), 
+          correctSide = Math.abs(calcAngleBetweenVectors(hatOrthogonal, bisectorVector)) >= PiBy2 ? 1 : -1, 
+          orthogonal = scaleHatVector(hatOrthogonal, s * correctSide),
+          proj1 = applySkew(A.add(orthogonal));
+        coords.push({
+          "projectedPoint": proj1,
+          "originPoint": A,
+          "bisector": bisector
+        });
+      });
+
+      // The points furthest from the vertex in the direction of the X and Y axes after distortion
+      // TODO: consider only projections that are inside the beginning and end of the circle segment
```

Comment:
not clear to me

Reply:
This is the case explained here: https://github.com/fabricjs/fabric.js/pull/8083#issuecomment-1264450123

`I'm considering the two points for projection, this can be a problem when the control points are too close. I'll adjust to consider only the point belonging to the yellow segment.`